src/notification.h

```diff
@@ -2,6 +2,13 @@
 #ifndef DUNST_NOTIFICATION_H
 #define DUNST_NOTIFICATION_H
 
+//TODO: Remove cyclic dependency
```

Comment:
I think the only thing we can do for the cyclical dependency is convert the `NotificationColor` refernces in settings.h to pointers and forward-define the typedef.

Reply:
Yes, this was my thought, too. But like the macro problem of `do { actual code } while (0)`, I think here is also a best solution applicable. And I think we should strive for consistency fix it to not tap into this again.

Reply:
Also, I cannot forward define the type, as I don't save the type as a pointer to a type's allocated memory. So, when defining `settings_t`, the size of `NotificationsColors` has to be known. 🤦‍